app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -128,12 +128,45 @@ def process_collection_item(item, key)
       new_result
     end
 
+    def find_host_by_provider_id(provider_id)
+      scheme, instance_uri = provider_id.split("://", 2)
+      prov, name_field = {
+        "gce" => [ManageIQ::Providers::Google::CloudManager, :name],
+        "aws" => [ManageIQ::Providers::Amazon::CloudManager, :uid_ems],
+      }[scheme]
+      instance_id = instance_uri.split('/').last
+
+      prov::Vm.find_by(name_field => instance_id) if !prov.nil? && !instance_id.blank?
+    end
+
+    def find_host_by_bios_uuid(new_result)
+      uuid_provider_types = [ManageIQ::Providers::Redhat::InfraManager::Vm.name,
+                             ManageIQ::Providers::Openstack::CloudManager::Vm.name,
+                             ManageIQ::Providers::Vmware::InfraManager::Vm.name]
+      identity_system = new_result[:identity_system].try(:downcase)
+      Vm.find_by(:uid_ems => identity_system,
+                 :type    => uuid_provider_types) if identity_system
+    end
+
+    def cross_link_node(new_result)
+      # Establish a relationship between this node and the vm it is on (if it is in the system)
+      if new_result[:identity_infra]
+        host_instance = find_host_by_provider_id(new_result[:identity_infra])
+      end
+      unless host_instance
```

Comment:
@simon3z  It is being tested (line 550 on refresh_parser_spec.rb) and it passes because host_instance will be nil and we do ```.try(:id)``` which works with nil. (lines 160-161 below). both ```find_host_by_*``` will return nil if they can't find any host.